### PR TITLE
R4R:add heart beat in cron task

### PR DESCRIPTION
**Pull request by weichang-bianjie**

### CODE REVIEW

File: backend/task/heartbeat.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+				return
+			}
+		}
+	}
+	doTask()
+	for {
+		select {
+		case <-ticker.C:
+			doTask()
+			logger.Info("HeartBeat is ok", logger.String("taskName", taskName))
```
Comment (kaifei-bianjie) [inline backend/task/heartbeat.go:62]: remove this log, too frequently

File: backend/task/task_control.go [hunk 2]
```diff
@@ -113,7 +113,8 @@ func (s TaskControlMonitor) Start() {
 	})
 }
 
-func (s TaskControlMonitor) DoTask() error {
+func (s TaskControlMonitor) DoTask(fn func(string) chan bool) error {
```
Comment (kaifei-bianjie) [inline backend/task/task_control.go:116]: how to judge whether a locked task should be unlocked

File: backend/task/asset_gateway.go [hunk 3]
```diff
@@ -25,7 +25,8 @@ func (task UpdateAssetGateways) Start() {
 	})
 }
 
-func (task UpdateAssetGateways) DoTask() error {
+func (task UpdateAssetGateways) DoTask(fn func(string) chan bool) error {
+	defer HeartQuit(fn(task.Name()))
```
Comment (kaifei-bianjie) [inline backend/task/asset_gateway.go:29]: where to invoke `heartBeat ` in doTask func

File: backend/task/task_control.go [hunk 4]
```diff
@@ -82,7 +82,8 @@ func (s TaskControlService) assetTaskShouldNotBeExecuted(taskName string, timeIn
 		}
 
 		now := time.Now().Unix()
-		if now-v.LatestExecTime < int64(timeInterval) && !isCronjob {
+		timeHeartBeat := conf.Get().Server.CronTimeHeartBeat
```
Comment (kaifei-bianjie) [inline backend/task/task_control.go]: what do you mean about these code?

Comment (weichang-bianjie) [reply backend/task/task_control.go]: have adjust code position

**Pull request merged**